### ISSUE TITLE
Load config only once

### ISSUE THREAD

**Issue by lkatalin** [keylime/rust-keylime#164]
From @ashcrow on #161 : I think we should eventually rework the configuration system so that we load the configuration once and then reload when we receive the proper signal. My reasoning is that the current set up means that if the file changes while the agent is running then the result could be old and new variables in use based on when they have been read.

**Comment by lkatalin** [comment]
These are two separate issues: 1) only load it once in the code and pass reference to config obj around; 2) reload if conf file is updated.

**Comment by lkatalin** [comment]
Closing in favor of #390
